docs/developer-guide/api.md

```diff
@@ -1,22 +1,24 @@
 ---
 layout: page
-title: ksqlDB REST API Reference
-tagline: Run queries over REST
+title: ksqlDB HTTP API Reference
+tagline: Run queries over HTTP
 description: Learn how to communicate with ksqlDB by using HTTP
 ---
 
 - [Get the Status of a ksqlDB Server (/info endpoint)](ksqldb-rest-api/info-endpoint.md)
 - [Run a ksqlDB Statement (/ksql endpoint)](ksqldb-rest-api/ksql-endpoint.md)
-- [Run A Query And Stream Back The Output (/query endpoint)](ksqldb-rest-api/query-endpoint.md)
+- [Execute a push or pull query and stream the results back (/query-stream endpoint)](ksqldb-rest-api/streaming-endpoint.md)
+- [Stream inserts to the server (/inserts-stream endpoint)](ksqldb-rest-api/streaming-endpoint.md)
```

Comment:
nit: can we link to the subheadings on this page for each of these two, so users don't have to additionally figure out where to go on the page? Or we can have intro text at the top of the streaming-endpoint.md file with links to each of the subheadings, similar to the intro on this page. Or both :)

Reply:
The old /query implementation is very inefficient - it has a scheduled executor which polls the queue. If we don't deprecate this endpoint right now (because of HTTP 1.1 only clients) I would suggest extending the new query-stream endpoint so that if the protocol is HTTP1.x it sends back the response using chunked encoding. This would be fairly straightforward to do. Then the old endpoint can be deprecated and removed eventually.